hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputSplit.java

```diff
@@ -444,4 +444,16 @@ public void setFormat(FileFormat fileFormat) {
   public Blocklet makeBlocklet() {
     return new Blocklet(getPath().getName(), blockletId);
   }
+
+  public String[] preferredLocations() {
```

Comment:
The super FileSplit.file is not serializable. Refer HADOOP-13519 so java serialization may return empty

Reply:
fixed